### ISSUE TITLE
 calico/node is not ready: BIRD is not ready: BGP not established(K8S 1.15.1,calico/cni:v3.8.1)

### ISSUE THREAD

**Issue by fivehundredmiless** [projectcalico/calico#2785]
#1 install
   I installed the k8s cluster with kubadm and the Pod CIDR Network is (192.168.0.0/16),The k8s cluster have 2 master and 1 node,These three nodes all in subnet (172.30.17.37/24).
I follow the below steps install the calico.

curl https://docs.projectcalico.org/v3.8/manifests/calico.yaml -O && kubectl apply -f calico.yaml

#2 problems

There is one node can not establish with others.

![image](https://user-images.githubusercontent.com/49806192/62709244-92bf1d00-ba27-11e9-9522-107e95bdb722.png)

describe pod as below .

![image](https://user-images.githubusercontent.com/49806192/62710246-8fc52c00-ba29-11e9-910f-c1e01c5d9aa9.png)

Error from server (NotFound): pods "pod" not found
 kubectl logs calico-node-7fhtp -n kube-system
2019-08-08 13:26:44.096 [INFO][8] startup.go 256: Early log level set to info
2019-08-08 13:26:44.096 [INFO][8] startup.go 272: Using NODENAME environment for node name
2019-08-08 13:26:44.096 [INFO][8] startup.go 284: Determined node name: xiucai-ops
2019-08-08 13:26:44.098 [INFO][8] k8s.go 228: Using Calico IPAM
2019-08-08 13:26:44.098 [INFO][8] startup.go 316: Checking datastore connection
2019-08-08 13:26:44.107 [INFO][8] startup.go 340: Datastore connection verified
2019-08-08 13:26:44.107 [INFO][8] startup.go 95: Datastore is ready
2019-08-08 13:26:44.113 [INFO][8] customresource.go 100: Error getting resource Key=GlobalFelixConfig(name=CalicoVersion) Name="calicoversion" Resource="GlobalFelixConfigs" error=the server could not find the requested resource (get GlobalFelixConfigs.crd.projectcalico.org calicoversion)
2019-08-08 13:26:44.121 [INFO][8] startup.go 584: Using autodetected IPv4 address on interface br-8225eb47243b: 172.18.0.1/16
2019-08-08 13:26:44.121 [INFO][8] startup.go 647: No AS number configured on node resource, using global value
2019-08-08 13:26:44.121 [INFO][8] startup.go 149: Setting NetworkUnavailable to False
2019-08-08 13:26:44.137 [INFO][8] startup.go 536: CALICO_IPV4POOL_NAT_OUTGOING is true (defaulted) through environment variable
2019-08-08 13:26:44.137 [INFO][8] startup.go 796: Ensure default IPv4 pool is created. IPIP mode: Always, VXLAN mode: Never
2019-08-08 13:26:44.162 [INFO][8] startup.go 806: Created default IPv4 pool (192.168.0.0/16) with NAT outgoing true. IPIP mode: Always, VXLAN mode: Never
2019-08-08 13:26:44.162 [INFO][8] startup.go 530: FELIX_IPV6SUPPORT is false through environment variable
2019-08-08 13:26:44.174 [INFO][8] startup.go 181: Using node name: xiucai-ops
2019-08-08 13:26:44.207 [INFO][20] k8s.go 228: Using Calico IPAM
2019-08-08 13:26:44.233 [INFO][20] ipam.go 87: Auto-assign 1 ipv4, 0 ipv6 addrs for host 'xiucai-ops'
2019-08-08 13:26:44.238 [INFO][20] ipam.go 313: Looking up existing affinities for host handle="ipip-tunnel-addr-xiucai-ops" host="xiucai-ops"
2019-08-08 13:26:44.251 [INFO][20] ipam.go 371: Ran out of existing affine blocks for host handle="ipip-tunnel-addr-xiucai-ops" host="xiucai-ops"
2019-08-08 13:26:44.262 [INFO][20] ipam.go 436: No more affine blocks, but need to allocate 1 more addresses - allocate another block handle="ipip-tunnel-addr-xiucai-ops" host="xiucai-ops"

**1. I have reinstall the calico serveal times and every time is the same node have problem.the other nodes can establish each other.**
**2. I use the problem node telnet to other nodes 179 port and the Connection is closed by foredgn host.**
---------------------------------------------------------
telnet 172.30.17.60 179
Trying 172.30.17.60...
Connected to 172.30.17.60.
Escape character is '^]'.
Connection closed by foreign host.

**3. I  try tcpdump on the normal nodes print as below :**

tcpdump -i eth0 port 179 -nnv
tcpdump: listening on eth0, link-type EN10MB (Ethernet), capture size 262144 bytes
22:15:10.942242 IP (tos 0xc0, ttl 64, id 47257, offset 0, flags [DF], proto TCP (6), length 60)
    172.30.17.60.53921 > 172.18.0.1.179: Flags [S], cksum 0x699c (incorrect -> 0x85f4), seq 1276858485, win 29200, options [mss 1460,sackOK,TS val 2377902316 ecr 0,nop,wscale 7], length 0
22:15:11.944102 IP (tos 0xc0, ttl 64, id 47258, offset 0, flags [DF], proto TCP (6), length 60)
    172.30.17.60.53921 > 172.18.0.1.179: Flags [S], cksum 0x699c (incorrect -> 0x820a), seq 1276858485, win 29200, options [mss 1460,sackOK,TS val 2377903318 ecr 0,nop,wscale 7], length 0
----------------------------------------------------
I try tcpdump the 179 port on problem nodes and got nothing.

tcpdump -i eth0 port 179 -nnv
tcpdump: listening on eth0, link-type EN10MB (Ethernet), capture size 262144 bytes

**4. process as blow**   
ps -ef | grep bird
root      6416  6358  0 22:13 ?        00:00:00 runsv bird6
root      6417  6358  0 22:13 ?        00:00:00 runsv bird
root      6553  6416  0 22:13 ?        00:00:00 bird6 -R -s /var/run/calico/bird6.ctl -d -c /etc/calico/confd/config/bird6.cfg
root      6554  6417  0 22:13 ?        00:00:00 bird -R -s /var/run/calico/bird.ctl -d -c /etc/calico/confd/config/bird.cfg
root     10947 13283  0 22:17 pts/1    00:00:00 grep --color=auto bird

I try to find the config file on each node all print as blow,is the normal ?
    ls /etc/calico/confd/config/bird.cfg
    ls: cannot access /etc/calico/confd/config/bird.cfg: No such file or directory

Thank you for your advice.

## Your Environment

* Calico version : calico/cni:v3.8.1
*  kubernetes,: K8S 1.15.1,
* Operating System Centos 7.5

**Comment by tmjd** [comment]
You said
> These three nodes all in subnet (172.30.17.37/24)

but the logs show
>2019-08-08 13:26:44.121 [INFO][8] startup.go 584: Using autodetected IPv4 address on interface br-8225eb47243b: 172.18.0.1/16

So I'm wondering if you have multiple interfaces on these hosts and if the one is selecting the wrong IP address to use. Perhaps you need to change the autodetection to ensure the correct IP gets selected, see https://docs.projectcalico.org/v3.8/reference/node/configuration#ip-setting

**Comment by fivehundredmiless** [comment]
> You said
> 
> > These three nodes all in subnet (172.30.17.37/24)
> 
> but the logs show
> 
> > 2019-08-08 13:26:44.121 [INFO][8] startup.go 584: Using autodetected IPv4 address on interface br-8225eb47243b: 172.18.0.1/16
> 
> So I'm wondering if you have multiple interfaces on these hosts and if the one is selecting the wrong IP address to use. Perhaps you need to change the autodetection to ensure the correct IP gets selected, see https://docs.projectcalico.org/v3.8/reference/node/configuration#ip-setting

As you said, thank you.

**Comment by tmjd** [comment]
Sounds like this has unblocked you, I'm going to close this.

**Comment by storm2013ak** [comment]
@fivehundredmiless What did you do to fix this issue? I'm having a similar issue where one of my calico nodes are not ready. I have a Kubernetes cluster with 3 master nodes. I can provide more information.

**Comment by storm2013ak** [comment]
![image](https://user-images.githubusercontent.com/19988992/86964567-e18ba000-c12b-11ea-9ca4-92882161614a.png)

`kubectl describe pod calico-node-brrkp`
![image](https://user-images.githubusercontent.com/19988992/86964642-06801300-c12c-11ea-96c8-66b1b9f58921.png)

I ran `kubectl logs calico-node-brrkp | grep "bird"` on a broken pod and I noticed this. I don't see this in other 2 calico node pods.
![image](https://user-images.githubusercontent.com/19988992/86964768-36c7b180-c12c-11ea-97de-7b7d5d98c3ef.png)

I also tried pinging the servers from another server and every packet seem to deliver. 

I ran this command as well `sudo calicoctl node status` from different servers. 
On the server that is "broken" I get this 
![image](https://user-images.githubusercontent.com/19988992/86965146-b5bcea00-c12c-11ea-958d-5fdb82c82c72.png)

On the other ones I get:
![image](https://user-images.githubusercontent.com/19988992/86965264-ddac4d80-c12c-11ea-97bb-46c70fc27145.png)

Any help is appreciated. I think all of this happened after I did apt upgrade on the server that is "broken" now. It upgraded the docker package, kubernetes-cni, also Linux kernel. I already reverted them back to what they were previously, hoping that it fixed my problem but it didn't.

I'm not sure If I should open another issue or not? Please let me know
